Fix 1215 - fix double nonworking expando

edhager opened the pull request:
The double expando icon problem was fixed in 0f67fa108b3952069ab36dd0cd0ad7fcbddf4a8e.  That fix avoided rendering the icon twice but did nothing to avoid registering multiple event handlers.  With two event handlers, the first would expand the clicked-on row and the second one would collapse it (`expand()` is a toggle if the second parameter is not passed).

kfranqueiro commented:
I've been working on refining this, and eventually discovered what seems like another bug we may have never noticed (unless I'm doing something really silly somehow): for some reason, the child levels' items are not properly indented.  `level` is seemingly always `0` when `renderCell` is processed for the children.  I suspect we're failing to pass `options` along somewhere again.

I could log this as a separate issue if you want, but either way, could you take a look?  I've pushed my own refined `kfranqueiro/1215-double-expando` branch that you can work from if you want.  I'll also leave a couple of comments on this original PR regarding stuff I've resolved on that branch, FYI.

kfranqueiro commented on test/data/index.json:
```diff
@@ -215,6 +215,12 @@
+		"title": "Test Grid Compound Columns with Tree",
+		"parent": "extensions"
+	},
+	{
```
FYI, you can regenerate this automatically by running `node test/data/createIndex.js`.  It will pick up the title from the `title` tag in the file automatically.

kfranqueiro commented on test/extensions/CompoundTree.html:
```diff
@@ -0,0 +1,85 @@
+
+		<script>
+			require(["dgrid/Tree", "dgrid/ColumnSet", "dgrid/OnDemandGrid",
+						"dgrid/extensions/CompoundColumns", "dstore/Memory"],
```
I've replaced `Memory` and the data on this page with `createHierarchicalStore` and `hierarchicalCountryData` like what's used in `Tree.html`.  Incidentally that's also what made the `level`-always-zero issue more apparent to me, since it's more obvious with nested parents.

kfranqueiro commented on test/intern/functional/TreeCompound.html:
```diff
@@ -0,0 +1,92 @@
+<html>
+<head>
+<meta charset="utf-8">
+<title>Test dgrid/selector module</title>
```
Oops. We apparently have a couple of copypasta'd `title` tags in the functional tests.  I need to do a lint sweep anyway so I'll hit the others then.  I've fixed this one on my branch though.

kfranqueiro commented on test/extensions/CompoundTree.html:
```diff
@@ -0,0 +1,85 @@
+		<meta name="viewport" content="width=570">
+		<style>
+			@import "../../../dojo/resources/dojo.css";
+			@import "../../css/dgrid.css";
```
This was missing a skin, which I've added on my branch.

edhager commented:
I added a commit to your branch and force pushed to my branch so the original diffs here are outdated.

The indent problem was due to ColumnSet not supporting the options object in createRowCells.  I fixed that and I added a test.  We already had an indent checking test in test/intern/mixins/Tree.js for OnDemandGrid + Tree.  That module is pretty long so I moved the indent test into its own module so I could add tests for OnDemandGrid + tree + compound columns + column sets.

edhager commented:
Fixes #1215

kfranqueiro commented:
These fixes have been merged to master (https://github.com/SitePen/dgrid/commit/26a6d00932a1af23a318bd4fafe280e0aa4b31a0 and https://github.com/SitePen/dgrid/commit/e69f1df17fc54c567def194169f6e63d4aad522e) and backported to dev-0.4 (https://github.com/SitePen/dgrid/commit/59a5589e981bfa0bcf86fcabd6b9a6feec1878a5 and https://github.com/SitePen/dgrid/commit/81c5a879801a4283c67d6596bb257b62b624acd0). Thanks!
